### PR TITLE
MC-21435: Replace styleName with className in kitchensink, use clsx where necessary.

### PR DESCRIPTION
**Pull request by sanmiawi**

### CODE REVIEW

File: kitchensink/package-lock.json [hunk 1]
```diff
@@ -3234,9 +3234,9 @@
       }
     },
     "classnames": {
-      "version": "2.2.6",
-      "resolved": "https://registry.npmjs.org/classnames/-/classnames-2.2.6.tgz",
-      "integrity": "sha512-JR/iSQOSt+LQIWwrwEzJ9uk0xfN3mTVYMwt1Ir5mUcSN6pU+V4zQFFaJsclJbPuAUQH+yfWef6tm7l1quW3C8Q=="
+      "version": "2.3.1",
```
Comment (justmejulian) [inline kitchensink/package-lock.json:3237]: This looks wrong 🤔

Comment (justmejulian) [reply kitchensink/package-lock.json:3237]: Could you try and run `npm run clean` and then `npm install` again?

**Review by justmejulian — APPROVED** [review]
Looks very good!
Just the package-lock.json looks a bit fishy

File: kitchensink/package-lock.json (hunk 1 above)
Comment (sanmiawi) [reply kitchensink/package-lock.json:3237]: will do

Comment (justmejulian) [reply kitchensink/package-lock.json:3237]: never mind. like discussed it is used by other libraries

**Comment by justmejulian** [comment]
There have been some changes to main since this started. Please merge main back into here and make sure there are still no usages of styleName or classnames.

- [ ] merge main into feature/MC-21435
- [ ] replace usages of styleName
- [ ] replace usages of classnames

File: kitchensink/src/app/common/PageElements/IconSelect/IconSelect.jsx [hunk 2]
```diff
@@ -21,8 +21,8 @@ const Option = (props) => {
 const IconSelect = (props) => {
   const options = icons.map((icon) => ({ value: icon.name, label: icon.name }));
   return (
-    <div styleName={classNames('selectContainer')}>
-      {props.title && <div styleName={classNames('selectTitle', { required: props.required && !props.disabled })}>{props.title}</div>}
+    <div className={clsx(styles.selectContainer)}>
```
Comment (Sakai-san) [inline kitchensink/src/app/common/PageElements/IconSelect/IconSelect.jsx:24]: you do not need `clx` here.
```
<div className={styles.selectContainer}>
```

**Review by Sakai-san — CHANGES_REQUESTED** [review]

**Review by Sakai-san — APPROVED** [review]

File: kitchensink/src/app/common/PageElements/IconSelect/IconSelect.jsx (hunk 2 above)
Comment (sanmiawi) [reply kitchensink/src/app/common/PageElements/IconSelect/IconSelect.jsx]: ✅

**Review by justmejulian — CHANGES_REQUESTED** [review]
This commit looks very broken. Let's try that again together

**Review by justmejulian — APPROVED** [review]

**Pull request merged**